pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -614,11 +622,19 @@ def get_result(self, handle: ResultHandle, **kwargs: KwargTypes) -> BackendResul
                         sleep(10)
 
                 res = job.result(timeout=kwargs.get("timeout", None))
-            assert isinstance(res, SamplerResult)
-            for circ_index, (r, d) in enumerate(zip(res.quasi_dists, res.metadata)):
-                self._ibm_res_cache[(jobid, circ_index)] = Counter(
-                    {n: int(0.5 + d["shots"] * p) for n, p in r.items()}
-                )
+            if isinstance(res, SamplerResult):
+                # TODO Is this code still reachable?
```

Comment:
Do we need to address this TODO here?

Reply:
I think it can wait; I will raise a new issue.

Reply:
#336